### PR TITLE
Add boost.scope_exit

### PR DESCRIPTION
**Pull request by hofbi**
Took the dependencies from https://github.com/nelhage/rules_boost/blob/e9bd4c17c976fa4ee0f3889d497e1d333f1f95d5/boost.BUILD#L1758-L1770

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (boost.scope_exit) have been updated in this PR. Please review the changes.

**Comment by hofbi** [comment]
The CI pipeline fails [here](https://buildkite.com/bazel/bcr-presubmit/builds/8118#01927978-5411-4e96-aa84-5d636dc5e4a5/147-156) complaining about an unstable URL where I get archive from. The error message suggests to use a github release, however this does not exists. Looking at other boost modules such as [static_assert](https://github.com/bazelbuild/bazel-central-registry/blob/3fa20c21280fa357079cdc9c6eb38c1f27f13aa7/modules/boost.static_assert/1.83.0/source.json#L4) shows they are using the same pattern. How could this work before?

**Comment by fmeum** [comment]
@wep21 @Vertexwahn

File: modules/boost.scope_exit/1.83.0/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+)
+
+bazel_dep(name = "boost.config", version = "1.83.0")
+bazel_dep(name = "boost.detail", version = "1.83.0")
+bazel_dep(name = "boost.function", version = "1.83.0")
+bazel_dep(name = "boost.mpl", version = "1.83.0")
+bazel_dep(name = "boost.preprocessor", version = "1.83.0")
+bazel_dep(name = "boost.type_traits", version = "1.83.0")
+bazel_dep(name = "boost.typeof", version = "1.83.0")
+bazel_dep(name = "boost.utility", version = "1.83.0")
```
Comment (wep21) [inline modules/boost.scope_exit/1.83.0/MODULE.bazel:15]:
```suggestion
bazel_dep(name = "rules_cc", version = "0.0.13")
```
missing rules_cc dependency

File: modules/boost.scope_exit/1.83.0/overlay/bazel-bin [hunk 2]
```diff
@@ -0,0 +1 @@
+/home/markus/.cache/bazel/_bazel_markus/0b53f018ebae478f2dceca6f76de4bf9/execroot/_main/bazel-out/k8-fastbuild/bin
```
Comment (wep21) [inline modules/boost.scope_exit/1.83.0/overlay/bazel-bin:1]: Please delete bazel-* files.

**Pull request closed without merging**